src/org/labkey/snd/SNDManager.java

```diff
@@ -2693,6 +2707,55 @@ public Event updateEvent(Container c, User u, Event event, boolean validateOnly)
         return event;
     }
 
+    // default to isUpdate = true
+    public int updateNarrativeCache(Container container, User user, Set<Integer>cacheData, Logger log)
+    {
+        return updateNarrativeCache(container, user, cacheData, log, true);
+    }
+    /* deletes and updates cached narratives */
+    public int updateNarrativeCache(Container container, User user, Set<Integer>cacheData, @NotNull Logger log, boolean isUpdate)
+    {
+        if (cacheData.size() > 0)
+        {
+            log.info("Deleting affected narrative cache rows.");
+            List<Map<String, Object>> rows = new ArrayList<>();
+            Map<String, Object> row;
+            BatchValidationException errors = new BatchValidationException();
+
+            for (Integer cacheDatum : cacheData)
+            {
+                row = new HashMap<>();
+                row.put("EventId", cacheDatum);
+                rows.add(row);
+            }
```

Comment:
You could move this for-loop down to the else statement.  Not needed anywhere else now.

Reply:
Good catch! I'll move it.